File: app-components/src/main/java/org/synsystems/onlypass/components/cryptography/random/SecureByteArrayGeneratorModule.java
package org.synsystems.onlypass.components.cryptography.random;

import org.synsystems.onlypass.components.AppScope;

import java.security.SecureRandom;

import dagger.Module;
import dagger.Provides;

@Module
@AppScope
public class SecureByteArrayGeneratorModule {
  @Provides
  @AppScope
  public SecureByteArrayGenerator provideSecureByteArrayGenerator(final SecureRandom secureRandom) {
    return new SecureByteArrayGenerator(secureRandom);
  }
}

File: app-components/src/main/java/org/synsystems/onlypass/components/cryptography/random/SecureByteArrayGeneratorModule.kt
package org.synsystems.onlypass.components.cryptography.random

import dagger.Module
import dagger.Provides
import org.synsystems.onlypass.components.AppScope
import java.security.SecureRandom

@Module
class SecureByteArrayGeneratorModule {
  @Provides
  @AppScope
  fun provideSecureByteArrayGenerator(secureRandom: SecureRandom) = SecureByteArrayGenerator(secureRandom)
}